source/common/network/hash_policy.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "envoy/network/hash_policy.h"
+#include "envoy/type/hash_policy.pb.h"
+
+#include "common/common/hash.h"
+
+namespace Envoy {
+namespace Network {
+/**
+ * Implementation of HashPolicy that reads from the proto TCP proxy config.
+ */
+class HashPolicyImpl : public Network::HashPolicy {
+public:
+  explicit HashPolicyImpl(absl::Span<const envoy::type::HashPolicy* const> hash_policy);
+
+  // Network::HashPolicy
+  absl::optional<uint64_t>
+  generateHash(const Network::Address::Instance* downstream_addr,
+               const Network::Address::Instance* upstream_addr) const override;
+
+  class HashMethod {
+  public:
+    virtual ~HashMethod() = default;
+    virtual absl::optional<uint64_t>
+    evaluate(const Network::Address::Instance* downstream_addr,
+             const Network::Address::Instance* upstream_addr) const PURE;
+  };
+
+  using HashMethodPtr = std::unique_ptr<HashMethod>;
+
+private:
+  std::vector<HashMethodPtr> hash_impls_;
```

Comment:
Since this is currently checked by schema, let's just store a single value here and assert that the passed in list of policies is of size 1.